src/main/java/org/akip/service/ProcessTimelineDefinitionService.java

```diff
@@ -0,0 +1,121 @@
+
+    @Autowired
+    private ProcessDeploymentRepository processDeploymentRepository;
+
+    @Autowired
+    private ProcessTimelineDefinitionRepository processTimelineDefinitionRepository;
+
+    private final ProcessTimelineDefinitionMapper processTimelineDefinitionMapper;
+
+    public ProcessTimelineDefinitionService(ProcessTimelineDefinitionMapper processTimelineDefinitionMapper) {
+        this.processTimelineDefinitionMapper = processTimelineDefinitionMapper;
+    }
+
+    public List<ProcessTimelineDefinitionDTO> getByBpmnProcessDefinitionId(String bpmnProcessDefinitionId) {
+        List<ProcessTimelineDefinitionDTO> processTimelineDefinitions = findAllByProcessDefinitionBpmnProcessDefinitionId(
+            bpmnProcessDefinitionId
+        );
+
+        if (processTimelineDefinitions.isEmpty()) {
+            throw new BadRequestErrorException("Timeline not defined for the process");
+        }
+
+        return processTimelineDefinitions;
+    }
+
+    public ProcessTimelineDefinitionDTO save(
+        String processDefinitionBusinessKey,
+        ProcessTimelineDefinitionDTO processTimelineDefinitionDTO
+    ) {
+        AtomicInteger stepCount = new AtomicInteger();
+        Optional<ProcessDefinition> processDefinition = processDefinitionRepository.findByBpmnProcessDefinitionId(
+            processDefinitionBusinessKey
+        );
+        ProcessTimelineDefinition processTimelineDefinition = processTimelineDefinitionMapper.toEntity(processTimelineDefinitionDTO);
+        processTimelineDefinition
+            .getItems()
+            .forEach(
+                item -> {
+                    item.setProcessTimelineDefinition(processTimelineDefinition);
+                    item.setStep(stepCount.getAndIncrement());
+                }
+            );
+        processTimelineDefinition.setProcessDefinition(processDefinition.get());
+        return processTimelineDefinitionMapper.toDto(processTimelineDefinitionRepository.save(processTimelineDefinition));
+    }
+
+    @Transactional(readOnly = true)
+    public List<ProcessTimelineDefinitionDTO> findAllByProcessDefinitionBpmnProcessDefinitionId(String processDefinitionBusinessKey) {
+        return processTimelineDefinitionRepository
+            .findAllByProcessDefinitionBpmnProcessDefinitionId(processDefinitionBusinessKey)
+            .stream()
+            .map(processTimelineDefinitionMapper::toDto)
+            .collect(Collectors.toList());
+    }
+
+    @Transactional(readOnly = true)
+    public Optional<ProcessTimelineDefinitionDTO> findOne(Long timelineDefinitionId) {
+        return processTimelineDefinitionRepository.findById(timelineDefinitionId).map(processTimelineDefinitionMapper::toDto);
+    }
+
+    public void delete(Long timelineDefinitionId) {
+        processTimelineDefinitionRepository.deleteById(timelineDefinitionId);
+    }
+
+    public List<TaskInstanceDTO> getBpmnUserTasks(String processDefinitionBusinessKey) {
```

Comment:
I'm not convinced that it is a good idea to include this method here. It is making the class less cohesive. One alternative is to include an util class. 
Another thing: there no property called `processDefinitionBusinessKey`. Do you mean `processDefinitionBpmnId`?